<launch>
  <!-- Tug ship -->
  <include file="$(find collision_avoidance_in_singapore)/launch/default_ship.launch">
    <arg name="waypoint_file" value="$(find asv_simulator)/config/waypoints/south_to_north.yaml" />
      <arg name="vessel_param_file" value="$(find collision_avoidance_in_singapore)/config/tug_punitha.yaml"/>
      <arg name="common_param_file" value="$(find collision_avoidance_in_singapore)/config/common.yaml"/>
      <arg name="loss_guidance_file" value="$(find collision_avoidance_in_singapore)/config/tug_punitha_lossguidance.yaml"/>
  </include>

  <!-- Obstacle ship 1 -->
  <include file="$(find collision_avoidance_in_singapore)/launch/default_ship.launch">
    <arg name="waypoint_file" value="$(find asv_simulator)/config/waypoints/north_to_south.yaml" />
    <arg name="vessel_param_file" value="$(find collision_avoidance_in_singapore)/config/vessel_punitha.yaml"/>
    <arg name="common_param_file" value="$(find collision_avoidance_in_singapore)/config/common.yaml"/>
    <arg name="loss_guidance_file" value="$(find collision_avoidance_in_singapore)/config/vessel_punitha_lossguidance.yaml"/>
    
    <arg name="use_vo" value="False" />
    <arg name="use_sb_mpc" value="False" />
    <arg name="vessel_model_file" value="$(find asv_simulator)/config/models/ship1.urdf" />
    <arg name="namespace" value="obstacles/ship1" />
    <arg name="shipname" value="ship1" />
    <arg name="initial_state" value="[0.0, 300.0, -1.57, 0.,0.,0.]" />
  </include>

  
  <!-- Obstacle ship 2 -->
  <!--
  <include file="$(find collision_avoidance_in_singapore)/launch/default_ship.launch">
    <arg name="waypoint_file" value="$(find collision_avoidance_in_singapore)/config/waypoints/crossed_north_to_south.yaml" />
    <arg name="vessel_param_file" value="$(find collision_avoidance_in_singapore)/config/vessel_punitha.yaml"/>
    <arg name="common_param_file" value="$(find collision_avoidance_in_singapore)/config/common.yaml"/>
    <arg name="loss_guidance_file" value="$(find collision_avoidance_in_singapore)/config/vessel_punitha_lossguidance.yaml"/>
    
    <arg name="use_vo" value="False" />
    <arg name="use_sb_mpc" value="False" />
    <arg name="vessel_model_file" value="$(find asv_simulator)/config/models/ship2.urdf" />
    <arg name="namespace" value="obstacles/ship2" />
    <arg name="shipname" value="ship2" />
    <arg name="initial_state" value="[70.0, 450.0, -1.57, 0.,0.,0.]" />
  </include>
  -->

  <include file="$(find asv_obstacle_tracker)/launch/default.launch"/>

  <node pkg="rviz"
        type="rviz"
        name="rviz"
        args="-d $(find collision_avoidance_in_singapore)/config/rviz/two_ships.rviz" />

  <!-- <node pkg="rosbag"
        type="record"
        name="rosbag_record"
        args="-O /home/ingerbha/ros_records/bagfiles/head_on.bag /asv/state /obstacles/ship1/state /obstacle_states"
        /> -->
</launch>
